spec/integration/data_stream/manifest.spec.yml

```diff
@@ -349,6 +348,64 @@ spec:
                             minProperties: 1
                             additionalProperties:
                               $ref: "#/definitions/elasticsearch_index_template/properties/mappings/properties/dynamic_templates/items/patternProperties/^_embedded_ecs/properties/mapping"
+            subobjects:
+              description: >
+                Turn the support for subobjects on or off in the mapping
+                configuration for the data stream.
+                
+                By default, it's set to `true`, enabling Elasticsearch to
+                expands fields that contain dots in their names
+                to their corresponding object structure.
+
+                Setting it to `false` restricts objects
+                from holding subobjects, enabling storing documents
+                with dot-contained field names and common prefixes.
+                
+                It is still possible to send documents that have
+                objects to a data stream that has set subobjects
+                to `false`.
+
+                Example when the subobjects mapping option is `false`:
+
+                PUT my-index-000001
+                {
+                  "mappings": {
+                    "subobjects": false 
+                  }
+                }
+
+                PUT my-index-000001/_doc/metric_1
+                {
+                  "time" : "100ms", 
+                  "time.min" : "10ms",
+                  "time.max" : "900ms"
+                }                
+
+                Before disabling subobjects, consider the following implications:
+
+                - Nested field types cannot be used in data streams.
+                - The subobjects mapping definition is immutable.
+                - This setting depends on auto-flattening mappings,
+                  which comes with limitations [^1] for integration
+                  and custom mappings in data streams without
+                  subobjects.
+                - Auto-flattening is available starting from version
+                  8.14, integrations must be on at least this
+                  version.
```

Comment:
Nit. This block of text (list) could be rendered with an unexpected format:
- Using this command to check how it is rendered:
  ```shell
  cat spec/integration/data_stream/manifest.spec.yml | yq -r .spec.definitions.elasticsearch_index_template.properties.mappings.properties.subobjects.description
  ```
This is how it is rendered (at least using `yq`):
```
Before disabling subobjects, consider the following implications:
- Nested field types cannot be used in data streams. - The subobjects mapping definition is immutable. - This setting depends on auto-flattening mappings,
  which comes with limitations [^1] for integration
  and custom mappings in data streams without
  subobjects.
- Auto-flattening is available starting from version
  8.14, integrations must be on at least this
  version.

```

Could it be added newlines or indentation. More or less it would have the same result

- Using newlines:
    - As there are already indented lines, there is no need to add a new one between the third and fourth item.

```suggestion
                Before disabling subobjects, consider the following implications:

                - Nested field types cannot be used in data streams.

                - The subobjects mapping definition is immutable.

                - This setting depends on auto-flattening mappings,
                  which comes with limitations [^1] for integration
                  and custom mappings in data streams without
                  subobjects.
                - Auto-flattening is available starting from version
                  8.14, integrations must be on at least this
                  version.
```
- Result
```
Before disabling subobjects, consider the following implications:
- Nested field types cannot be used in data streams.
- The subobjects mapping definition is immutable.
- This setting depends on auto-flattening mappings,
  which comes with limitations [^1] for integration
  and custom mappings in data streams without
  subobjects.
- Auto-flattening is available starting from version
  8.14, integrations must be on at least this
  version.
```

- Using indentation
```suggestion
                Before disabling subobjects, consider the following implications:

                  - Nested field types cannot be used in data streams.
                  - The subobjects mapping definition is immutable.
                  - This setting depends on auto-flattening mappings,
                    which comes with limitations [^1] for integration
                    and custom mappings in data streams without
                    subobjects.
                  - Auto-flattening is available starting from version
                    8.14, integrations must be on at least this
                    version.
```

- Result

```
Before disabling subobjects, consider the following implications:

  - Nested field types cannot be used in data streams.
  - The subobjects mapping definition is immutable.
  - This setting depends on auto-flattening mappings,
    which comes with limitations [^1] for integration
    and custom mappings in data streams without
    subobjects.
  - Auto-flattening is available starting from version
    8.14, integrations must be on at least this
    version.
```

Reply:
Great point!

Personally, I prefer the indentation-based solution, trying thing one!